.github/workflows/ci.yml

```diff
@@ -137,11 +201,33 @@ jobs:
       - name: Build ${{ env.PACKAGE_NAME }}
         run: |
           echo "${{ secrets.GITHUB_TOKEN }}" | docker login docker.pkg.github.com -u awslabs --password-stdin
-          export DOCKER_IMAGE=docker.pkg.github.com/awslabs/aws-iot-device-client/amazonlinux:latest
+          export DOCKER_IMAGE=docker.pkg.github.com/awslabs/aws-iot-device-client/ubuntu-16-x64:latest
           docker pull $DOCKER_IMAGE
-          docker run --mount type=bind,source=$(pwd),target=/root/${{ env.PACKAGE_NAME }} --env GITHUB_REF $DOCKER_IMAGE
+          docker run --mount type=bind,source=$(pwd),target=/root/${{ env.PACKAGE_NAME }} --env GITHUB_REF $DOCKER_IMAGE --compile-mode=shared_lib_mode
+      - name: Archive ubuntu-16-x64 Build Artifact                                                                                                                                                        
+        uses: actions/upload-artifact@v2                                                                                                                                                                      
+        # Run for main branch only                                                                                                                                                                            
+        if: github.ref == 'refs/heads/main'                                                                                                                                                                   
+        with:                                                                                                                                                                                                 
+          name: "SL.ubuntu.x64.${{ env.PACKAGE_NAME }}"                                                                                                                          
```

Comment:
I think this naming convention is confusing - the static builds get DC for Device Client but the shared library builds get SL for shared library. I think all of them should get DC to differentiate between ST builds and then somewhere in the artifact name you can just append "shared", with static builds just not having the extension, or having "static"